app/lib/common/widgets/share/action/share_space_object_action.dart

```diff
@@ -167,4 +219,25 @@ class ShareSpaceObjectActionUI extends StatelessWidget {
       _ => null,
     };
   }
+
+  Future<RefDetails?> getRefDetails({
+    required WidgetRef ref,
+    required SpaceObjectDetails objectDetails,
+  }) async {
+    final objectId = objectDetails.objectId;
+    switch (objectDetails.objectType) {
+      case ObjectType.pin:
+        final sourcePin = await ref.watch(pinProvider(objectId).future);
+        return await sourcePin.refDetails();
+      case ObjectType.calendarEvent:
+        final sourcePin =
+            await ref.watch(calendarEventProvider(objectId).future);
+        return await sourcePin.refDetails();
+      case ObjectType.taskList:
+        final sourcePin = await ref.watch(taskListProvider(objectId).future);
+        return await sourcePin.refDetails();
```

Comment:
```suggestion
        final source = await ref.watch(taskListProvider(objectId).future);
        return await source.refDetails();
```

not a pin.

these are clearly copy paste, another indicator that this API probably doesn't need this match...

Reply:
`another indicator that this API probably doesn't need this match..`

Sorry not getting you on this, can you explain more on this?

Reply:
this is in reference to the open discussion above.